recursion/core/src/memory/air.rs

```diff
@@ -0,0 +1,155 @@
+use core::mem::size_of;
+use p3_air::{Air, BaseAir};
+use p3_field::PrimeField32;
+use p3_matrix::dense::RowMajorMatrix;
+use p3_matrix::MatrixRowSlices;
+use sp1_core::air::{AirInteraction, SP1AirBuilder};
+use sp1_core::lookup::InteractionKind;
+use sp1_core::{air::MachineAir, utils::pad_to_power_of_two};
+use std::borrow::Borrow;
+use std::borrow::BorrowMut;
+
+use super::columns::MemoryInitCols;
+use crate::air::Word;
+use crate::memory::MemoryChipKind;
+use crate::memory::MemoryGlobalChip;
+use crate::runtime::ExecutionRecord;
+
+pub(crate) const NUM_MEMORY_INIT_COLS: usize = size_of::<MemoryInitCols<u8>>();
+
+// pub(crate) const MEMORY_INIT_COL_MAP: MemoryInitCols<usize> = make_col_map();
+
+// const fn make_col_map() -> MemoryInitCols<usize> {
+//     let indices_arr = indices_arr::<NUM_MEMORY_INIT_COLS>();
+//     unsafe { transmute::<[usize; NUM_MEMORY_INIT_COLS], MemoryInitCols<usize>>(indices_arr) }
+// }
+
+#[allow(dead_code)]
+impl MemoryGlobalChip {
```

Comment:
It would be nice if we could have only a single memory chip generic in the word but maybe that is not necessary and even bad if RISV-AIR and Recursion Air are completely separate crates